src/main/java/org/opensearch/knn/index/codec/util/KNNCodecUtil.java

```diff
@@ -102,4 +106,20 @@ public static long getTotalLiveDocsCount(final BinaryDocValues binaryDocValues)
         }
         return totalLiveDocs;
     }
+
+    public static List<String> getEngineFiles(String extension, String fieldName, SegmentInfo segmentInfo) {
```

Comment:
please add java docs on the public function.

and also unit test for this function.

Reply:
> please add java docs on the public function.

LGTM

Reply:
FIXED at https://github.com/opensearch-project/k-NN/pull/1946/commits/3b5751c08732d0a147880627fd44ae8b3196fe98

Reply:
Having a separate UT will be better KNNWeight now will not be the only consumer of this function

Reply:
Added at https://github.com/opensearch-project/k-NN/pull/1946/commits/439c5b7e01f4bf1d75c8f9e369c1674be21f2d50#